--- libs/database/prisma/seed.ts
+++ libs/database/prisma/seed.ts
@@ -107,31 +107,36 @@
   console.log('Upserted Programs');
 
   // 4. Academic Years
+  const today = new Date();
+  const startYear = today.getMonth() >= 6 ? today.getFullYear() : today.getFullYear() - 1; // July is 6
+
+  function getAyName(year: number) {
+    const nextYearStr = (year + 1).toString().slice(-2);
+    return `${year}-${nextYearStr}`;
+  }
+
   const academicYearsData = [
     {
-      name: '2025-26',
-      startDate: new Date('2025-07-01'),
-      endDate: new Date('2026-06-30'),
+      name: getAyName(startYear - 1),
+      startDate: new Date(`${startYear - 1}-07-01`),
+      endDate: new Date(`${startYear}-06-30`),
       isActive: false,
     },
     {
-      name: '2026-27',
-      startDate: new Date('2026-07-01'),
-      endDate: new Date('2027-06-30'),
+      name: getAyName(startYear),
+      startDate: new Date(`${startYear}-07-01`),
+      endDate: new Date(`${startYear + 1}-06-30`),
       isActive: true,
     },
     {
-      name: '2027-28',
-      startDate: new Date('2027-07-01'),
-      endDate: new Date('2028-06-30'),
+      name: getAyName(startYear + 1),
+      startDate: new Date(`${startYear + 1}-07-01`),
+      endDate: new Date(`${startYear + 2}-06-30`),
       isActive: false,
     },
   ];
 
   const academicYears: Record<string, any> = {};
